### PR TITLE
modules: hal_nordic: Don't get DT Node when temp update is disabled

### PR DESCRIPTION
**Pull request by FilipZajdel**
Avoid getting DT Node of temperature sensor when temperature update
is disabled.

Signed-off-by: Filip Zajdel <filip.zajdel@nordicsemi.no>

### CODE REVIEW

File: modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c [hunk 1]
```diff
@@ -35,7 +35,11 @@ struct temperature {
 
 static struct temperature temperature = {
 	.value = DEFAULT_TEMPERATURE,
-	.dev = DEVICE_DT_GET(DT_NODELABEL(temp))
+#if defined(CONFIG_NRF_802154_TEMPERATURE_UPDATE)
+	.dev = DEVICE_DT_GET(DT_NODELABEL(temp)),
+#else
+	.dev = NULL,
+#endif
 };
```
[the remark this answers is not in the document]
Comment (FilipZajdel) [reply modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c:43]: It probably would be easier, but `nrf_802514_temperature_get` is a common function for sensor and sensorless implementations. If it looks better I can put the structure instantiation under common `#if defined(CONFIG_NRF_802154_TEMPERATURE_UPDATE)` and change `nrf_802154_temperature_get` to have two implementations: one for case when temperature is updated and the second one for the opposite.

[the remark this answers is not in the document]
Comment (jciupis) [reply modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c:43]: `nrf_802154_temperature_get` can still have a single, common implementation that returns a static variable initialized with `DEFAULT_TEMPERATURE`. The data needed to update that variable as time goes by would by defined only if `CONFIG_NRF_802154_TEMPERATURE_UPDATE`, as @anangl suggested. Am I missing something?

File: modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c [hunk 2]
```diff
@@ -25,47 +25,41 @@
...
-
-static struct temperature temperature = {
-	.value = DEFAULT_TEMPERATURE,
-	.dev = DEVICE_DT_GET(DT_NODELABEL(temp))
-};
 
 #if defined(CONFIG_NRF_802154_TEMPERATURE_UPDATE)
+
+static const struct device *temperature_dev = DEVICE_DT_GET(DT_NODELABEL(temp));
+static struct k_work_delayable temperature_work;
```
Comment (jciupis) [inline modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c:34]: nit: it's pretty clear from the context that all data in this file exists for the purpose of temperature readouts, so IMO `temperature_` prefix is redundant. We can only leave `value`, `dev` and `work`.

**Review by jciupis — DISMISSED** [review]
One minor, non-blocking comment. The PR looks good to me.

**Review by anangl — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by jciupis — DISMISSED** [review]

**Review by hubertmis — APPROVED** [review]
This change looks good to me.
Out of curiosity: what's the intended usage of this feature? When the temperature update could be disabled?

**Review by anangl — APPROVED** [review]

**Comment by FilipZajdel** [comment]
@hubertmis Disabling the temperature update will be useful when the board does not feature the temperature sensor, for example a simulated board.

**Review by nordic-krch — APPROVED** [review]

**Comment by hubertmis** [comment]
> for example a simulated board.

Do you mean a partial simulation which does not simulate internal temperature sensor?

**Pull request merged**